tests/manage/monitoring/conftest.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import pytest
+import threading
+import time
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.utility.prometheus import PrometheusAPI
+
+
+logger = logging.getLogger(__name__)
+
+
+def measure_operation(
+    operation, minimal_time=None, metadata=None, measure_after=False
+):
+    """
+    Get dictionary with keys 'start', 'stop', 'metadata' and 'result' that
+    contain information about start and stop time of given function and its
+    result.
+
+    Args:
+        operation (function): Function to be performed.
+        minimal_time (int): Minimal number of seconds to run, it can be more
+            based on given operation.
+        metadata (dict): This can contain dictionary object with information
+            relevant to test (e.g. volume name, operating host, ...).
+        measure_after (bool): Determine if time measurement is done before or
+            after the operation returns its state. This can be useful e.g.
+            for capacity utilization testing where operation fills capacity
+            and utilized data are measured after the utilization is completed.
+
+    Returns:
+        dict: contains information about `start` and `stop` time of given
+            function and its `result` and provided `metadata`.
+    """
+    def prometheus_log(info, alert_list):
+        """
+        Log all alerts from Prometheus API every 10 seconds.
+
+        Args:
+            run (bool): When this var turns into False the thread stops.
+            alert_list (list): List to be populated with alerts
+        """
+        prometheus = PrometheusAPI()
+        while info.get('run'):
+            alerts_response = prometheus.get(
+                'alerts',
+                payload={
+                    'silenced': False,
+                    'inhibited': False
+                }
+            )
+            assert alerts_response.ok, 'Prometheus API request failed'
+            for alert in alerts_response.json().get('data').get('alerts'):
+                if alert not in alert_list:
+                    logger.info(f"Adding {alert} to alert list")
+                    alert_list.append(alert)
+            time.sleep(10)
```

Comment:
why sleep?
Please try to avoid using sleep and try to sample for a state you are waiting for

Reply:
This is part of a function that is not waiting for any alerting status. It is just looking periodically for alerts but it waits for signal from main thread to stop so it doesn't make sense to have sampling function here.

Reply:
ok now I can see it.
So can we at least sample more periodically? 3 seconds maybe?

Reply:
done